### ISSUE TITLE
Allow for locking of the database. 

### ISSUE THREAD

**Issue by nrxus** [prisma/prisma#3088]
## Problem

I need to be able to create a row in a table based on information of whether a similar row exists or not:

1. Lock table for creation
2. Check if similar row exists
3. If similar row exists, create it and mark it as similar. Else, create it and mark it as brand new.
4. Unlock

I need to be able to lock the table such that steps 1 and 2 happen "atomically". i.e., if another request comes in to create that row, then there will be no race condition between the checks of these two requests. 

Ideally there would be a way to lock only creation based on some other criteria, but that seems to be outside of the scope. 

## Suggested solution
```ts
prisma.lock(MyTable) { |t|
  t.checkMyStuff()
  t.createMyStuff()
}
```

## Alternatives
```ts
prisma.myTable.lock(|t| {})
```

```ts
prisma.atomic(|p| => {
  // do stuff with the atomic version of the prisma client
})
```

I am open to anything else that would fix the core issue. 

## Additional context
I know there is a lot debate about transactions. This does not need a long-living transaction, just lock.

**Comment by matthewmueller** [comment]
Hey @nrxus 

Can you elaborate on what `t.checkMyStuff()` would do? Is it a database call, a network request?

**Comment by nrxus** [comment]
It is a database call.

**Comment by 000dry** [comment]
did this go anywhere? I would imagine there are plenty of users who would benefit from the ability to lock tables

**Comment by janpio** [comment]
No, the feature request is still open so it did not get implemented.
